aten/src/ATen/cuda/detail/IndexUtils.cuh

```diff
@@ -13,7 +13,7 @@ TORCH_CUDA_API bool canUse32BitIndexMath(const at::Tensor &t, int64_t max_elem=s
 
 template <typename scalar, typename IndexType>
 TensorInfo<scalar, IndexType>
-getTensorInfo(const at::Tensor& t) {
+getTensorInfo(const at::Tensor& t, bool positiveDim = false) {
```

Comment:
Please add a comment describing what the positiveDim bool does

Reply:
Having had a chance to read the entire PR, this is the one change I'm still curious about. Can you describe your thinking here? I believe I understand what's going on but would like to verify.

Reply:
I have added a comment:

```
// Argument `positiveDim` turns on the legacy behavior of THC's getTensorInfo:
// if `t` is a scalar, it's interpreted as a single-element vector.
// (See the legacy implementation in aten/src/THC/THCTensorTypeUtils.cuh)
```

Unfortunately I didn't note down the details, but I remember that some tests were failing without handling this special case.

Reply:
The special case this is handling is when the tensor is a zero-dim "scalar tensor," right?

Reply:
yes, correct

Reply:
Got it. Instead of extending `getTensorInfo`, what would you think about checking for a 0-dim tensor and just calling `torch.full` to handle that case? Is that too involved a change at this point?

Reply:
I don't think it's too involved but TBH I don't understand how you would like me to use torch.full.